File: http/src/main/java/webconnect/com/webconnect/observer/SuccessLiveData.java
package webconnect.com.webconnect.observer;


import android.arch.lifecycle.MutableLiveData;

public class SuccessLiveData extends MutableLiveData<String> {

    private static SuccessLiveData sInstance;

    private SuccessLiveData() {

    }

    public static SuccessLiveData getInstance() {
        if (sInstance == null) {
            synchronized (SuccessLiveData.class) {
                if (sInstance == null) {
                    sInstance = new SuccessLiveData();
                }
            }
        }
        return sInstance;
    }
}

File: http/src/main/java/webconnect/com/webconnect/observer/SuccessLiveData.kt
package webconnect.com.webconnect.observer;


import android.arch.lifecycle.LiveData
import android.arch.lifecycle.MutableLiveData;

object SuccessLiveData {

    internal val success = MutableLiveData<String>()

    @JvmStatic
    fun getSuccessLiveData(): LiveData<String> {
        return success
    }
}